products-api-rest/src/Crud/ProductsCrud.php

```diff
@@ -43,11 +45,23 @@ public function findAll(): array
     return ($products === false) ? [] : $products;
   }
 
+  
   public function find(int $id): ?array
   {
-    return [];
+      if ($id === 0) {
+          throw new InvalidArgumentException("L'ID spécifié n'est pas valide.");
+      }
+      $query = "SELECT * FROM products WHERE id = :id";
+      $stmt = $this->pdo->prepare($query);
+      $stmt->execute(['id' => $id]);
+      if (!$stmt->rowCount()) {
+          throw new OutOfBoundsException("L'ID spécifié n'existe pas.");
+      }
+      $item = $stmt->fetch();
+      return $item ?? null;
```

Comment:
Je pense voir ce que tu veux faire ici, mais dans ce cas pourquoi utiliser l'opérateur NULL coalescent pour retourner null si `$item` est...null ? 😅
À la place, si tu veux retourner un truc par défaut si `$item` est false, je pense que tu peux utiliser la version courte de la ternaire (non, on l'a pas vue en cours ) :
```suggestion
      return $item ?: null;
```
https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.ternary
"Expression expr1 ?: expr3 evaluates to the result of expr1 if expr1 evaluates to true, and expr3 otherwise"